community/index/src/test/java/org/neo4j/index/internal/gbptree/SeekCursorTestBase.java

```diff
@@ -1199,7 +1199,7 @@ public void mustRereadHeadersOnRetry() throws Exception
         // WHEN
         try ( SeekCursor<KEY,VALUE> cursor = new SeekCursor<>( this.cursor,
                 node, from, to, layout, stableGeneration, unstableGeneration, () -> 0L, failingRootCatchup,
-                unstableGeneration, exceptionDecorator ) )
+                unstableGeneration, exceptionDecorator, 1 ) )
```

Comment:
If we are only testing with `maxCacheSize=1` we are not actually testing this new behaviour. Ideally we would test with some different cache sizes. Maybe even `random(1,20)`.
Some tests would need to be updated for this, must notably `mustFindKeyInsertedOnSeekPosition`.